src/renderer/components/currency/CurrencyInfo.tsx

```diff
@@ -1,21 +1,67 @@
-import React from 'react'
+import React, { useState, useMemo, useCallback } from 'react'
 
-import { bn, assetAmount, formatAssetAmountCurrency } from '@xchainjs/xchain-util'
+import { assetAmount, formatAssetAmountCurrency } from '@xchainjs/xchain-util'
+import { Row, Dropdown } from 'antd'
 import BigNumber from 'bignumber.js'
 import * as O from 'fp-ts/lib/Option'
 import { pipe } from 'fp-ts/pipeable'
 
+import { SlipTolerance } from '../../../shared/api/types'
+import { ReactComponent as DownIcon } from '../../assets/svg/icon-down.svg'
 import { sequenceTOption } from '../../helpers/fpHelpers'
+import { ChangeSlipToleranceHandler } from '../../services/app/types'
 import { PoolAssetDetail } from '../../services/midgard/types'
 import * as Styled from './CurrencyInfo.styles'
 
 type CurrencyInfoProps = {
   from?: O.Option<PoolAssetDetail>
   to?: O.Option<PoolAssetDetail>
   slip?: BigNumber
+  slipTolerance: SlipTolerance
+  changeSlipTolerance: ChangeSlipToleranceHandler
 }
 
-export const CurrencyInfo = ({ to = O.none, from = O.none, slip = bn(0) }: CurrencyInfoProps) => {
+const SLIP_PERCENTAGES = [3, 5, 10]
+
+export const CurrencyInfo = ({ to = O.none, from = O.none, slipTolerance, changeSlipTolerance }: CurrencyInfoProps) => {
+  const [slipSettingsVisible, setSlipSettingsVisible] = useState(false)
+
+  const changeSlipToleranceHandler = useCallback(
+    (slipTolerance) => {
+      changeSlipTolerance(slipTolerance as SlipTolerance)
```

Comment:
Then no need to type cast `... as SlipTolerance` I think... (see previous comment)